### PR TITLE
feat(core/inlines): allow excluding abbr from inlines

### LINKED ISSUE #3514: How to turn off the abbreviations feature?
**Is your feature request related to a problem? Please describe.**

I don't know how to turn off the abbreviations feature.

**Describe the solution you'd like**

It would be useful if there's a configuration option or something to tell ReSpec not to wrap abbreviations with `<abbr>` automatically.

**Additional context**

For some documents, I only want to mark some (but not all) abbreviations, and their expansions are not the same (even though the abbreviations are the same). However, as far as I understand ReSpec can’t handle this because it will automatically wrap all matching abbreviations with `<abbr>` (and I don't know which expansion ReSpec will choose).

This feature request is currently mainly for the word "GB" in [clreq](https://w3c.github.io/clreq/). I just want to add `abbr` for the "main text" (like "GB-style" in [3.1.4](https://w3c.github.io/clreq/#prohibition_rules_for_line_start_end)), but not for the standard code (like "GB/T 15834-2011"). Even if we (the editors of the document) decide to add `abbr` to all instances of this abbreviation, since the expansions in each language are different, it is troublesome to edit when adding it. And if there is one place that we forget to add, the (automatically added) expansion may be wrong.

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #3514 

One can now add an "exclude" CSS class to have an `<abbr>` excluded from processing in inlines.

### CODE REVIEW

File: src/core/inlines.js [hunk 1]
```diff
@@ -258,9 +258,11 @@ export function run(conf) {
 
   // PRE-PROCESSING
   /** @type {NodeListOf<HTMLElement>} */
-  const abbrs = document.querySelectorAll("abbr[title]");
-  for (const abbr of abbrs) {
-    abbrMap.set(abbr.textContent, abbr.title);
+  const abbrElements = document.querySelectorAll("abbr[title]:not(.exclude)");
```
Comment (sidvishnoi) [inline src/core/inlines.js:261]: We should document the `exclude` class while we're here.
It's also used with WebIDL blocks apparently: https://github.com/w3c/respec/blob/77ab0c7507d9960c50745dde954a87e199e2be81/tests/spec/core/idl-index-spec.js#L62

Comment (marcoscaceres) [reply src/core/inlines.js:261]: Correct. We might need to update our PR template so we don't forget to document things.

File: src/core/inlines.js [hunk 2]
```diff
@@ -258,9 +258,11 @@ export function run(conf) {
...
+  for (const { textContent, title } of abbrElements) {
+    const key = norm(textContent);
+    const value = norm(title);
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/inlines.js:264]: I felt the same... but then was thinking it doesn't do any harm as then we don't care if people are sloppy with their markup.

[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/inlines.js:264]: It's a slight hit on performance (too many small hits mean overall slowdown).

**Review by sidvishnoi — APPROVED** [review]
LGTM.
Remove norm(title) if you feel like.

File: src/core/inlines.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/inlines.js:264]: Can live with it :)

**Comment by xfq** [comment]
It seems to be documented in https://github.com/w3c/respec/wiki/exclude but not in https://respec.org/docs/. Is that intended?

**Comment by sidvishnoi** [comment]
Just merged and released https://github.com/w3c/respec-web-services/pull/196

**Pull request merged**